src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteFactory.cs

```diff
@@ -557,13 +609,19 @@ private static void VerifyOpenGenericAotCompatibility(Type serviceType, Type[] g
             }
         }
 
-        public void Add(Type type, ServiceCallSite serviceCallSite)
+        public void Add(ServiceIdentifier serviceIdentifier, ServiceCallSite serviceCallSite)
         {
-            _callSiteCache[new ServiceCacheKey(type, DefaultSlot)] = serviceCallSite;
+            _callSiteCache[new ServiceCacheKey(serviceIdentifier, DefaultSlot)] = serviceCallSite;
         }
 
-        public bool IsService(Type serviceType)
+        public bool IsService(Type serviceType) => IsService(new ServiceIdentifier(null, serviceType));
+
+        public bool IsService(object key, Type serviceType) => IsService(new ServiceIdentifier(key, serviceType));
```

Comment:
Do we need to add this to `IServiceProviderIsService`? I assume that the existing `IsService(Type)` method will always return false for a named service, is that right?

Reply:
Fixed